### ISSUE TITLE
Karpenter stuck with pod `Pod should schedule on: nodeclaim/...`

### ISSUE THREAD

**Issue by darklight147** [Azure/karpenter-provider-azure#432]
### Version

__Karpenter Version:__ v0.0.0

__Kubernetes Version:__ v1.0.0

### Expected Behavior

Create a new node

### Actual Behavior

<img width="1006" alt="image" src="https://github.com/user-attachments/assets/30d85204-01ff-4fe1-a532-e41a2a43125b">
Show the above message when describing a Pod but doesn't create any new Nodes

### Steps to Reproduce the Problem

AKS Cluster with node auto provisioning enabled 

Scale a deployment Nginx for example to 20 with memory request 8Gi

### Resource Specs and Logs

```yaml
apiVersion: karpenter.sh/v1beta1
kind: NodePool
metadata:
  annotations:
    karpenter.sh/nodepool-hash: "12393960163388511505"
    karpenter.sh/nodepool-hash-version: v2
    kubernetes.io/description: General purpose NodePool for generic workloads
    meta.helm.sh/release-name: aks-managed-karpenter-overlay
    meta.helm.sh/release-namespace: kube-system
  creationTimestamp: "2024-07-11T00:34:44Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
    helm.toolkit.fluxcd.io/name: karpenter-overlay-main-adapter-helmrelease
    helm.toolkit.fluxcd.io/namespace: 668f23a48709cf00012ccf73
  name: default
  resourceVersion: "485239"
  uid: 19ab7243-b704-4a1c-b9d8-279243f12865
spec:
  disruption:
    budgets:
    - nodes: 100%
    consolidationPolicy: WhenUnderutilized
    expireAfter: Never
  template:
    spec:
      nodeClassRef:
        name: default
      requirements:
      - key: kubernetes.io/arch
        operator: In
        values:
        - amd64
      - key: kubernetes.io/os
        operator: In
        values:
        - linux
      - key: karpenter.sh/capacity-type
        operator: In
        values:
        - on-demand
      - key: karpenter.azure.com/sku-family
        operator: In
        values:
        - D
status:
  resources:
    cpu: "16"
    ephemeral-storage: 128G
    memory: 106086Mi
    pods: "110"
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by Bryce-Soghigian** [comment]
Looking at the cluster id you shared, I see logs like `creating instance, insufficient capacity, regional on-demand vCPU quota limit for subscription has been reached. To scale beyond this limit, please review the quota increase process here: https://learn.microsoft.com/en-us/azure/quotas/regional-quota-requests` 

If you do `kubectl get events` | grep karp, do you see any events like this?

**Comment by Bryce-Soghigian** [comment]
You can unblock your scaleup by requesting additional quota on the subscription for that region via this link https://learn.microsoft.com/en-us/azure/quotas/regional-quota-requests

**Comment by darklight147** [comment]
Empty events from the command

<img width="560" alt="image" src="https://github.com/user-attachments/assets/851f5693-eed7-4334-a201-6d9a76e53ce8">

Also here is the current Quota sorted by Current usage

<img width="1476" alt="image" src="https://github.com/user-attachments/assets/f5f7a8b2-f393-4a2f-8515-e343f786f33c">

**Comment by darklight147** [comment]
@Bryce-Soghigian hey any update on this? thank you 🚀

**Comment by maulik13** [comment]
We are seeing a similar behaviour in our cluster. Node claims are created, but they do not get in to the ready state. We do not see anything special in the events. It only says "Pod should schedule on: nodeclaim/app-g5gcw" and "Cannot disrupt NodeClaim" for existing nodes. 

We have also checked that we have not reached our quota.

**Comment by maulik13** [comment]
We are still seeing this behavior time to time where nodeclaims are created resulting in creation of new VMs, but they do not manage to join the cluster and get in to a `Ready = true` state. How do we debug this or provide you with logs? The only solution is to reconcile the state by running an empty update against the cluster.

**Comment by superff** [comment]
is BootstrapToken token expired? that is a hard coded value in the value.yaml of the helm chat and It will expire

**Comment by tallaxes** [comment]
As discussed, based on symptoms described this is most likely related to token expiration / a duplicate of https://github.com/Azure/karpenter-provider-azure/issues/695. Closing as duplicate, please reopen (or create a new issue) if you observe something similar that is _not_ related to token expiration.
